lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -483,6 +477,58 @@ static void
   lookupTypeMembers(BaseTy, PT, Consumer, CurrDC, LS, Reason);
 }
 
+/// Trigger synthesizing implicit member declarations to make them "visible".
+static void synthesizeMemberDeclsForLookup(NominalTypeDecl *NTD,
```

Comment:
How is this triggered during normal type-checking? This seems kind of brittle to live in lookup-visible-decls, although admittedly this whole file is basically like that...

Reply:
If the type is declared in the primary files, they are synthesized in `TypeChecker::typeCheckDecl()`(`DeclChecker`) or in `typeCheckFunctionsAndExternalDecls()`.

But if it's in the non-primary files, AFAIU, they are lazily synthesized on demand in `DeclContext::lookupQualified()` 
https://github.com/apple/swift/blob/bd971507bae5a12a2b40365e795799037ec59fcc/lib/AST/NameLookup.cpp#L1573-L1579
https://github.com/apple/swift/blob/bd971507bae5a12a2b40365e795799037ec59fcc/lib/Sema/TypeChecker.h#L1018-L1024

I can move this function into `TypeChecker::`, but I don't think it's worth...